hcipy/aperture/realistic.py

```diff
@@ -349,6 +349,94 @@ def aper(grid):
 
 	return aper
 
+def make_luvoir_b_aperture(normalized=False, with_segment_gaps=True, gap_padding=1, segment_transmissions=1,
+						   return_header=False, return_segments=False):
+	'''
+	Segment gaps can be included or excluded, and the transmission for each
+	of the segments can also be changed. Segments can be returned as well.
+
+	Parameters
+	----------
+	normalized : boolean
+		If this is True, the pupil diameter will be scaled to 1. Otherwise, the
+		diameter of the pupil will be 15.0 meters.
+	with_segment_gaps : boolean
+		Include the gaps between individual segments in the aperture.
+	gap_padding : scalar
+		Arbitrary padding of gap size to represent gaps on smaller arrays - this effectively
+		makes the gaps larger and the segments smaller to preserve the same segment pitch.
+	segment_transmissions : scalar or array_like
+		The transmission for each of the segments. If this is a scalar, this transmission
+		will be used for all segments.
+	return_header : boolean
+		If this is True, a header will be returned giving all important values for the
+		created aperture for reference.
+	return_segments : boolean
+		If this is True, the segments will also be returned as a ModeBasis.
+
+	Returns
+	-------
+	aperture : Field generator
+		The LUVOIR B aperture.
+	aperture_header : dict
+		A dictionary containing all quantities used when making this aperture. Only returned if
+		`return_header` is True.
+	segments : list of Field generators
+		The segments. Only returned when `return_segments` is True.
+	'''
+	pupil_diameter = 8.0 #m actual circumscribed diameter, used for lam/D calculations other measurements normalized by this diameter
+	pupil_inscribed = 6.7 #m actual inscribed diameter
+	actual_segment_flat_diameter = 0.955 #m actual segment flat-to-flat diameter
+	actual_segment_gap = 0.006 #m actual gap size between segments
+
+	segment_gap = actual_segment_gap * gap_padding #padding out the segmentation gaps so they are visible and not sub-pixel
```

Comment:
See comment above about the equivalent LUVOIR A function.

Reply:
The one-line descriptions of hard-coded parameters (ie. actual_segment_gap and pupil_diameter) are fine btw, since that's mostly variable description rather than doing some calculation.